### PR TITLE
zutty: init at 0.16-unstable-2024-08-18

### PR DESCRIPTION
**Pull request by rolfschr**
## Description of changes

This PR adds [zutty](https://tomscii.sig7.se/zutty/), a terminal emulator.

It is based (and improves) on this previous [PR](https://github.com/NixOS/nixpkgs/pull/257156) which is now outdated. Compared to that, there are several improvements:
* A basic test is added
* `src` has been updated (on GitHub anymore)
* The "font path" issue [discussed there](https://github.com/NixOS/nixpkgs/pull/257156#discussion_r1380341764) has been addressed (see this [comment](https://github.com/NixOS/nixpkgs/pull/257156#issuecomment-2278807153), too)
* A newer version of zutty is used that contains a fix for certain fonts (see details [here](https://gist.github.com/rolfschr/4261ea856535c6007194b2b27f3dee04?permalink_comment_id=5161242#gistcomment-5161242))

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by s0me1newithhand7s** [comment]
i'll test on `x86_64-linux` asap, amd after OfBorg eval.

**Comment by rolfschr** [comment]
ofBorg failes with `error: evaluation aborted with the following error message: 'lib.customisation.callPackageWith: Function called without required argument "glew-egl" at /ofborg/checkout/1/repo/38dca4e3aa6bca43ea96d2fcc04e8229/mr-est/ofborg-evaluator-4/pkgs/by-name/zu/zutty/package.nix:10'`

I don't know what the problem is. Initially, I thought that this is because `glew-egl` is not supported on all platforms but zutty's package definition is for [linux only](https://github.com/NixOS/nixpkgs/pull/336255/files#diff-f74a8bad5debfaeb697a41681939b9ebe6704d885dbe7297a3449b4551d13b12R56) anyways.

**Comment by rolfschr** [comment]
Ah, I found the issue, I need to use `glew` instead of glew-egl. See [here](https://github.com/NixOS/nixpkgs/blob/8dd85c637c66ed74bb631486015b19914752c69b/pkgs/top-level/aliases.nix#L530).

**Comment by rolfschr** [comment]
ofBorg has okay'ed everything except "zutty.passthru.tests on aarch64-darwin" which will be skipped (darwin is [not supported](https://github.com/NixOS/nixpkgs/pull/336255/files#diff-f74a8bad5debfaeb697a41681939b9ebe6704d885dbe7297a3449b4551d13b12R49)). 

@s0me1newithhand7s This looks good to me now. Could you have a look into this, please?

@wegank You had checked https://github.com/NixOS/nixpkgs/pull/257156 so could you maybe approve & merge here? I think the other PR could then be closed.

**Comment by rolfschr** [comment]
@JJJollyjim You've been designing (I think) `nixos/tests/terminal-emulators.nix` so you might be a good candidate for review/merging. Would you mind doing so? Thanks.

**Comment by rolfschr** [comment]
@AndersonTorres You had previously merged a terminal emulator-related PR. Could you have a look into this and possibly merge?

File: pkgs/by-name/zu/zutty/package.nix [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+  wafHook,
+  python3,
+  libXmu,
+  glew,
+  ucs-fonts,
+  nixosTests,
+}:
+stdenv.mkDerivation (finalAttrs: {
+  pname = "zutty";
+  version = "0.16-1-g04b2ca3";
```
Comment (wegank) [inline pkgs/by-name/zu/zutty/package.nix:16]:
```suggestion
  version = "0.16-unstable-2024-08-18";
```

https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#versioning

**Review by s0me1newithhand7s — APPROVED** [review]
LGTM

**Comment by wegank** [comment]
Result of `nixpkgs-review pr 336255` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>zutty</li>
  </ul>
</details>

**Comment by s0me1newithhand7s** [comment]
congrats!

**Pull request merged**